### DISCUSSION TITLE
Problem installing Zowe CLI on fedora gnome

### DISCUSSION THREAD

**Discussion by Caseybu** [zowe/zowe-cli#1585] [General]
I can not get zowe cli to install
[root@fedora sdr]# npm install -g prebuild-install
/usr/local/bin/prebuild-install -> /usr/local/lib/node_modules/prebuild-install/bin.js
+ prebuild-install@7.1.1
updated 1 package in 8.024s
[root@fedora sdr]# npm install -g @zowe/cli@zowe-v2-lts

> @zowe/cli@7.9.2 preinstall /usr/local/lib/node_modules/@zowe/cli
> node ./scripts/preinstall

internal/fs/utils.js:332
    throw err;
    ^

Error: EACCES: permission denied, open '.prebuild-installrc'
    at Object.openSync (fs.js:497:3)
    at Object.writeFileSync (fs.js:1528:35)
    at Object.<anonymous> (/usr/local/lib/node_modules/@zowe/cli/scripts/preinstall.js:17:8)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:76:12)
    at internal/main/run_main_module.js:17:47 {
  errno: -13,
  syscall: 'open',
  code: 'EACCES',
  path: '.prebuild-installrc'
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @zowe/cli@7.9.2 preinstall: `node ./scripts/preinstall`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @zowe/cli@7.9.2 preinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2022-12-14T17_02_13_112Z-debug.log

**Comment by t1m0thyj** [comment]
Hello @Caseybu, this issue is related to root permissions being required to create files in the directory where npm installs global packages.

It may help to change to a different directory with more lax permissions - see [npm docs](https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally#manually-change-npms-default-directory) for how to do this. There is also a [sample script in the CLI repo](https://github.com/zowe/zowe-cli/blob/master/tools/linux-environment-setup.sh#L89-L156) that may be helpful to show how to change the "npm prefix" (directory where global packages are installed).

An alternative option may be to add the flag `--unsafe-perm=true` on your npm install command, if you are ok with allowing install scripts to run with root privileges.
